### PR TITLE
Log bridge rate limits

### PR DESCRIPTION
**Pull request by rudle**
I'm not sure what numbers to fill in here yet. Will need to deploy & observe the results of my previous PR to determine the constants.

### CODE REVIEW

File: pkg/logbridge/logbridge.go [hunk 1]
```diff
@@ -63,6 +80,17 @@ func (lb *LogBridge) LossyCopy(r io.Reader, capacity int) {
 	var n int
 	var err error
 	for line := range lines {
+		err = lb.logLineRateLimit.WaitN(context.TODO(), 1)
```
Comment (anthonybishopric) [inline pkg/logbridge/logbridge.go]: for maximum awesome, should we extract this fake context out of the for loop?

Comment (rudle) [inline pkg/logbridge/logbridge.go]: allocations in the loop all day long

File: pkg/logbridge/logbridge.go [hunk 2]
```diff
@@ -100,8 +129,9 @@ func (lb *LogBridge) lossyCopy(r io.Reader, lines chan []byte) {
 		case lines <- line:
 		default:
 			droppedLines++
+			lb.droppedLineCount.Inc(1)
```
Comment (anthonybishopric) [inline pkg/logbridge/logbridge.go]: Isn't this limit just based on the channel capacity and not the lovely rate limiter you've just added?

Comment (rudle) [inline pkg/logbridge/logbridge.go]: the rate limit will cause the channel (buffer) to fill up which will cause us to fall into this case

I agree that this isn't a precise metric, but I think it's better than nothing?

Comment (anthonybishopric) [inline pkg/logbridge/logbridge.go]: Discussed IRL, will switch to `ReserveN` in order to track a second gauge related to how long we were waiting

**Comment by rudle** [comment]
@anthonybishopric PTAL I've changed this to use `Reserve…` vs `Wait…`

**Comment by rudle** [comment]
Wondering how badly the histogram will become skewed in the case that the application under log stops logging to STDOUT. We can emit a heartbeat update of "0ms" via a separate go routine, but that would seem to imply synchronization inside logbridge's main loop (which is undesirable for performance reasons).

**Comment by rudle** [comment]
This uses a simple counter to address the concern I highlighted in: https://github.com/square/p2/pull/503#issuecomment-222839086

bonus: the tests are passing.

**Comment by anthonybishopric** [comment]
👍

**Comment by rudle** [comment]
thanks for the review. Gonna fixup the godeps (removing net/context :key:) squash and merge.

**Pull request merged**